pkg/cri/server/sandbox_run.go

```diff
@@ -222,12 +173,27 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 	if err != nil {
 		return nil, fmt.Errorf("failed to create containerd container: %w", err)
 	}
+
+	// Add container into sandbox store in INIT state.
+	sandbox.Container = container
```

Comment:
NOTING(non-blocker):

The containerd might restart at this point and the netns path will be emtpy.
After containerd restarts, the sanbox.NetNS is always [non-nil][1] an contains empty path. 
The behaviour is like what we do to host-networking. 

Just curious that if the `Stat/Lstat` changes and the empty path is allowed :). 
Maybe we can fast-return if it is empty path when [umount nspath][2]. Or we can set the sandbox.NsPath is nil if it is empty at the [statement][1].

[1]: <https://github.com/containerd/containerd/blob/3228848c6cf413d9026e187749300dc53149c0ef/pkg/cri/server/restart.go#L419>
[2]: <https://github.com/containerd/containerd/blob/release/1.6/pkg/netns/netns_linux.go#L131>

Reply:
`Stat/Lstat` returns "no such file" error if it is an empty path. 
I prefer setting `sandbox.NetNS` to nil in cri/server/restart.go

Reply:
This failed the `TestRunPodSandboxWithShimDeleteFailure` as `PodSandboxStatus` returns "error reading from server: EOF" error when we set `sandbox.NetNS` to nil. Need follow-up on this.